### PR TITLE
Always pass POSIX paths to pygit2 index (#2185)

### LINKED ISSUE #2185: BUG: failed git detection (despite it being on PATH) on windows
With 3.45.0, I get

```
ImportError: Failed to initialize: Bad git executable.
The git executable must be specified in one of the following ways:
    - be included in your $PATH
    - be set via $GIT_PYTHON_GIT_EXECUTABLE
    - explicitly set via git.refresh(<full-path-to-git-executable>)

All git commands will error until this is rectified.
```

It's definitely on my path though
```
>where git
C:\Users\[xxx]\.pixi\envs\git\Library\bin\git.exe
C:\Users\[xxx]\.pixi\bin\git.bat
```

### PR DESCRIPTION
**Pull request by mgorny**
Checklist
* [x] Added a ``news`` entry

libgit2 expects internal git paths to be passed in git convention, that is using forward slashes.  Use pathlib.Path.as_posix() to ensure that the correct conversion is used when passing paths to index operations.

Fixes #2185

### CODE REVIEW

**Comment by mgorny** [comment]
@h-vetinari, could you check if this helps with #2191 by any chance?

File: news/2185-fix-pygit2-windows.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+* <news item>
+
+**Removed:**
+
+* <news item>
+
+**Fixed:**
+
+* Fix passing paths to ``pygit2`` on Windows (#2185)
```
Comment (beckermr) [inline news/2185-fix-pygit2-windows.rst:19]:
```suggestion
* Fix passing paths to ``pygit2`` on Windows. (#2192)
```

We want the number of this PR, not what it fixes.

**Review by beckermr — APPROVED** [review]

**Review by h-vetinari — APPROVED** [review]
Confirmed to work, thank you very much. Even better, it also fixed #2191! 🥳 

Changes themselves LGTM as well!

File: conda_smithy/feedstock_io.py [hunk 2]
```diff
@@ -38,7 +39,10 @@ def set_exe_file(filename, set_exe=True):
 
     repo = get_repo(filename)
     if repo:
-        index_entry = repo.index[os.path.relpath(filename, repo.workdir)]
+        index_path = (
+            Path(filename).resolve().relative_to(repo.workdir).as_posix()
```
[the remark this answers is not in the document]
Comment (beckermr) [reply conda_smithy/feedstock_io.py:43]: It doesn't seem necessary to me. No need to test functionality that is built into python. Hopefully they do that.

[the remark this answers is not in the document]
Comment (beckermr) [reply conda_smithy/feedstock_io.py:43]: I'm going to merge but happy to add a test in a second PR if you feel strongly.

**Pull request merged**